### PR TITLE
Fix parametrize test cancel mount workspace

### PR DESCRIPTION
**Pull request by touilleMan**

### CODE REVIEW

File: tests/core/mountpoint/test_base.py [hunk 1]
```diff
@@ -579,22 +580,26 @@ async def test_cancel_mount_workspace(base_mountpoint, alice_user_fs, event_bus,
...
-    # doesn't go undetected.
-    timeout = 3.0 if sys.platform == "darwin" else 1.0
-    with trio.fail_after(timeout * 0.9):
+    # Reuse the same mountpoint manager for all the mountings to
+    # make sure state is not polutated by previous mount attempts
+    async with mountpoint_manager_factory(
+        alice_user_fs, event_bus, base_mountpoint
+    ) as mountpoint_manager:
+
+        for timeout in count(0, 0.002):
```
Comment (vxgmichel) [inline tests/core/mountpoint/test_base.py:589]: Using `itertools.count` plus a break once the workspace has been mounted is a very neat idea :+1:

File: tests/core/mountpoint/test_base.py [hunk 2]
```diff
@@ -579,22 +580,26 @@ async def test_cancel_mount_workspace(base_mountpoint, alice_user_fs, event_bus,
     """
     wid = await alice_user_fs.workspace_create("w")
 
-    # The timeout for `_stop_fuse_thread` is 1 second (3 seconds for macOS),
-    # so let's use a slightly lower timeout to make sure a potential failure
-    # doesn't go undetected.
-    timeout = 3.0 if sys.platform == "darwin" else 1.0
```
Comment (vxgmichel) [inline tests/core/mountpoint/test_base.py:585]: :facepalm:

File: parsec/core/mountpoint/fuse_runner.py [hunk 3]
```diff
@@ -251,9 +251,16 @@ async def _stop_fuse_thread(
...
+            # Restart the unmount process if necessary, this is needed in
+            # case the stop is ordered before fuse has finished started (hence
+            # the unmount command can run before the OS mount has occured).
+            # Of course this means in theory we could be umounting by mistake
+            # an unrelated mountpoint that tooks our path, but it's a really
+            # unlikely event.
             if process.poll() is not None:
                 process = await trio.open_process(process_args)
+        else:
+            logger.error("Fuse thread stop timeout", mountpoint=mountpoint_path)
```
Comment (vxgmichel) [inline parsec/core/mountpoint/fuse_runner.py:263]: :+1: on adding logger errors

**Review by vxgmichel — APPROVED** [review]
Good gatch, LGTM :+1:

File: tests/core/mountpoint/test_base.py (hunk 2 above)
Comment (touilleMan) [reply tests/core/mountpoint/test_base.py:585]: image

**Pull request merged**